src/sorting/selection_sort.rs

```diff
@@ -1,6 +1,10 @@
 pub fn selection_sort<T: Ord>(arr: &mut [T]) {
+    if arr.is_empty() {
+        return;
+    }
+
     let len = arr.len();
-    for left in 0..len {
+    for left in 0..(len - 1) {
```

Comment:
Although there is some additional code here, I believe it makes it easier for people to understand. The textbook's description of selection sort states that the first loop only needs to go up to `len - 1`. I think this modification will not cause confusion.

Reply:
The original code is simply shorter and handles all of the cases. I really do not see a reason why to introduce a new _branch_.

Reply:
@vil02 Thank you for your feedback. I agree that the original code is more succinct and covers all cases. Therefore, I have decided to close this PR. Thank you for your time and advice!